coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -620,15 +621,21 @@ class SingleClusterPlanner(val dataset: Dataset,
 
         if (nameFilter.isEmpty) (nameFilter, leFilter, lp)
         else {
-          val filtersWithoutBucket = rawSeriesLp.filters.filterNot(_.column.equals(PromMetricLabel)).
-            filterNot(_.column == "le") :+ ColumnFilter(PromMetricLabel,
-            Equals(nameFilter.get.replace("_bucket", "")))
-          val newLp =
-            if (lp.isLeft)
-             Left(lp.left.get.copy(rawSeries = rawSeriesLp.copy(filters = filtersWithoutBucket)))
-            else
-             Right(lp.right.get.copy(series = rawSeriesLp.copy(filters = filtersWithoutBucket)))
-          (nameFilter, leFilter, newLp)
+          // the convention for histogram bucket queries is to have the "_bucket" string in the suffix
+          if (!nameFilter.get.endsWith("_bucket")) {
+            (nameFilter, leFilter, lp)
+          }
+          else {
+            val filtersWithoutBucket = rawSeriesLp.filters.filterNot(_.column.equals(PromMetricLabel)).
+              filterNot(_.column == "le") :+ ColumnFilter(PromMetricLabel,
+              Equals(nameFilter.get.replace("_bucket", "")))
```

Comment:
Do we only replace the _bucket in the end?
consider this case
`time_bucket_bucket  => time_bucket`

Reply:
@yu-shipit excellent point :) I wanted to keep the PR as simple as I can, without breaking any existing scenarios. So I just added an additional check before the existing replace. 

But you make an excellent point. This is definitely a good case. 

 @alextheimer @amolnayak311 @sherali42 should we tackle this case that @yu-shipit pointed out ? Any other scenario that we should think for ?

Reply:
I went ahead and added a helper function to do this. Please let me know if there is any concern here.